sdk/go/pulumi/context.go

```diff
@@ -282,6 +293,229 @@ func (ctx *Context) IsConfigSecret(key string) bool {
 	return false
 }
 
+// registerTransform starts up a callback server if not already running and registers the given transform.
+func (ctx *Context) registerTransform(t XResourceTransform) (*pulumirpc.Callback, error) {
+	contract.Assertf(ctx.state.supportsTransforms, "transform attempted but not supported by resource monitor")
```

Comment:
Could this just be a regular error rather than an assert?  I the call [here](https://github.com/pulumi/pulumi/pull/15355/files#diff-6ac1847413cb55c8d480423da6e23ed6ac98c0066fcf7c57f8b5cd960a04dee1R2106) is protected by an error return, but could we end up [here](https://github.com/pulumi/pulumi/pull/15355/files#diff-6ac1847413cb55c8d480423da6e23ed6ac98c0066fcf7c57f8b5cd960a04dee1R1161) and run into this assert?  Or maybe that's not possible and I'm just not familiar enough with the engine yet?

Reply:
Yup moved the check